tests/qat/test_qasm.py

```diff
@@ -228,6 +228,46 @@ def test_cx_override(self):
             == 2
         )
 
+    def test_ecr_override(self):
+        hw = get_default_echo_hardware(4)
+        parser = Qasm3Parser()
+        result = parser.parse(get_builder(hw), get_qasm3("ecr_override_test.qasm"))
+        # tests that there are 2 extra_soft_square pulses from the custom def
+        assert (
+            len(
+                [
+                    inst
+                    for inst in result.instructions
+                    if hasattr(inst, "shape")
+                    and (inst.shape is PulseShapeType.EXTRA_SOFT_SQUARE)
+                ]
+            )
+            == 2
+        )
```

Comment:
same here

Reply:
I'm not sure what the the check from line 254 onwards adds doesn't it essentially check that hw.create_builder() creates a fresh builder which doesn't "hang on" to the overrides? 

Also the override tests (test_ecr_override, test_cx_override, test_cx_override) look like they could be parameterised by the gate? `get_qasm3(f"{gate}_override_test.qasm")` or similar (we could actually parameterise the .qasm files too with jinja2 etc. but I think that's for another day).

Reply:
The tests only override the gate for qubits (1, 2), so we want to make sure that gate on qubits (0, 1) and (2, 3) are unchanged, i.e., equivalent to what we will expect to see from the circuit builder. So we check that the instruction lists match.

Nice suggestion, on it now! :)